packages/analytics-browser/src/transports/xhr.ts

```diff
@@ -17,14 +17,12 @@ export class XHRTransport extends BaseTransport implements Transport {
       xhr.open('POST', serverUrl, true);
       xhr.onreadystatechange = () => {
         if (xhr.readyState === this.state.done) {
+          const responseText = xhr.responseText;
           try {
-            const responsePayload = xhr.responseText;
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-            const parsedResponsePayload: Record<string, any> = JSON.parse(responsePayload);
-            const result = this.buildResponse(parsedResponsePayload);
-            resolve(result);
-          } catch (e) {
-            reject(e);
```

Comment:
This is something different with v2.x.
Should we reject it when there have an error?

Reply:
main and v1.x branches should be the same. Please see the other [PR](https://github.com/amplitude/Amplitude-TypeScript/pull/758/files) in review where this logic is the same. 

We should not reject it because it's possible that this is a success response with no or non-json format body.

Reply:
If an error occurs in the try block, will we lose the useful information about why the error occurred?

Reply:
We use try catch block here because of `JSON.parse()`. I don't see anywhere else errors can occur.

Reply:
Got it.